contrib/nydus-test/functional-test/test_nydusify.py

```diff
@@ -80,11 +71,11 @@ def test_basic_conversion(
     workload_gen.finish_torture_read()
 
 
+@pytest.mark.skip(reason="build cache has bug trying use https to access repo")
```

Comment:
@bergwolf @changweige nydusify always use https to access source/target/build cache registry, and then fallback to plain http if https failure. I noticed that in one case this might not work and it has been fixed in this [commit](https://github.com/dragonflyoss/image-service/pull/966/commits/559e277a9f931516def19db3d13b8c3bd26e83f8), please take a look.

Reply:
@imeoer 

The problem still exists.

```
INFO[2022-12-30T11:13:18+08:00] [CACH] Check layer                            ChainID="sha256:0310f4cd9792b14930d36723933ce1811a4d0aeb1b5a57984ef49b0cdf8886e5" Time="956.344µs"
WARN[2022-12-30T11:13:18+08:00] Failed to get cache record: Push cached bootstrap layer: failed to do request: Head "https://127.0.0.1:5000/v2/nydus/golang/blobs/sha256:62d40f76fdad58cd102ad1f5f0567ae65d842c629425c429bb90c959c545c993": http: server gave HTTP response to HTTPS client
INFO[2022-12-30T11:13:18+08:00] [SOUR] Mount layer

```